### DISCUSSION TITLE
Patch for V380E21_V3.3.3.0_20201212

### DISCUSSION THREAD

**Discussion by hrpoint** [Arkady23/V380-Web#20] [General]
Hi there,
Great work, really need that!
Does it enable telnet access by deafult?

Which patch would I need for this cheap banggood camera: Guudgo 1080P 10 LED Outdoor PTZ 

Software version AppV380E21_V3.3.3.0_20201212
Firmware version HwV380E21_WF3_PTZ_WIFI_20201212

**Comment by Arkady23** [comment]
Try as indicated in the description httpd_V380E3_C5_v20210607.zip Hm.... In my opinion, there was no such camera yet. We need to find out the Hwname and version.

**Comment by hrpoint** [comment]
Hi, thnx for answering
I have tried to contact v380technical to get firmware but I got no answer.
Is there any other way to find out the Hwname and version?

**Comment by hrpoint** [comment]
Just got the answer from v380technical, firmware is here:
https://ufile.io/nwsan5fj

**Reply by Arkady23** [reply]
Is this a patch from your camera? Have you installed it? Is it installing? If it is installed, then this is a patch from your camera and means you have a camera, as I wrote above V380E3_C5. There is an http server for this camera in the table. Have you found it or should I give you a link?

**Reply by Arkady23** [reply]
https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_V380E3_C5_v20210607.zip

**Comment by hrpoint** [comment]
The file is the firmware I got from v380technical when I told them which firmware I currently have , are they trustworthy?
(My current firmware information is listed in the first post).

Is there any other way to find out whether they sent me the right firmware or I will brick the camera?

**Reply by Arkady23** [reply]
If there is an official latest firmware, then the application on your smartphone should prompt you and you can update your firmware. If you don't trust the firmware you've been given, then don't install it. I described all the ways to find out the camera model in the project. You can install my firmware with a web server completely safely. Please note that there is already an update to this version of the web server. But in any case, first you need to install a patch with the first version of the web server.

**Reply by hrpoint** [reply]
I will try it with your firmware.
(App says that my current firmware is the latest).

Thnx for the support - you are great 👍

**Comment by hrpoint** [comment]
It works!!! 

So for anyone buying Guudgo 1080P 10 LED Outdoor PTZ on Banggood, it is the V380E3_C5:
Camera Model	V380E3_C5
Processor Model	ARM926EJ-S rev 5 (v5l) Cloud39EV3_AK3918E80PIN_MNBD
CPU architecture	5TEJ
Firmware Version	3030207
Hardware Version	HwV380E21_WF6_PTZ_WIFI_20160217
App Version	AppV380E21_V3.3.2.7_20201112

Btw, Is it possible to get current jpg snapshot from web server? 
That would be great.

**Reply by Geotv123** [reply]
> It works!!!
> 
> So for anyone buying Guudgo 1080P 10 LED Outdoor PTZ on Banggood, it is the V380E3_C5: Camera Model V380E3_C5 Processor Model ARM926EJ-S rev 5 (v5l) Cloud39EV3_AK3918E80PIN_MNBD CPU architecture 5TEJ Firmware Version 3030207 Hardware Version HwV380E21_WF6_PTZ_WIFI_20160217 App Version AppV380E21_V3.3.2.7_20201112
> 
> Btw, Is it possible to get current jpg snapshot from web server? That would be great.

Hi @hrpoint 
Can you be so kind to provide the patch the v380tech support sent you, as well as the file Arkady23 sent you that worked? The v380E3_C5 file that is on the table now is not working with the exact camera like yours. Thank you 

the link from January says I need a paid account to download and The Arkady23 link is dead now

**Comment by Arkady23** [comment]
All active links are here https://github.com/Arkady23/V380-Web#%D1%82%D0%B0%D0%B1%D0%BB%D0%B8%D1%86%D0%B0-%D0%BE%D0%BF%D1%80%D0%B5%D0%B4%D0%B5%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-%D0%BC%D0%BE%D0%B4%D0%B5%D0%BB%D0%B8-%D0%BA%D0%B0%D0%BC%D0%B5%D1%80%D1%8B-v380

If you managed to install some other camera on the web server, please let me know so that I can add information to this table.

**Reply by Geotv123** [reply]
Hi Arkady, The V380E3_C5 from the table is not working on this camera or my other camera (Q12) from this thread where you provided the same file: https://github.com/bcaller/v380-ipcam-firmware-patch/issues/5#issuecomment-1113638107

On the other camera Q12 I left you my soft id and the patch I was sent from V380 support. Maybe you can help there and I will try to get the soft id for this one now.
Thank you for your help

**Reply by Arkady23** [reply]
What is Q12? Can you at least write the full title? For example, HwV380E21_WF6_PTZ_WI FI. By the way, try the [update for Q7](https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_V380E3_C5.YW_9090909_v20210620.zip), the full name is HwV380E22_WF3_MINI_Q7.

**Comment by Geotv123** [comment]
I apologize, I tried to include a link but it didn’t work. Here it is:  “M@cro-Video V380 Q12 Surveillance Camera” HwV380E22_WF6_CAT_PTZ_20160207
with {"soft_id":"V380E3_C5","dev_id":,"version":3030209,"timestamp":1651593582}  let me know if you need dev_id 

I just found the soft Id for for my camera like @hrpoint also HwV380E21_WF3_PTZ_WIFI_20201212
{“soft_id”:”V380E3_C5”,”dev_id”:,”version”:3030300,”timestamp”:1652148189}

**Comment by Arkady23** [comment]
I do not know what is the matter. Most likely, the official patch will not work for you either. It's probably some kind of special software.

**Reply by Geotv123** [reply]
> I do not know what is the matter. Most likely, the official patch will not work for you either. It's probably some kind of special software.

Thank you for trying to help. I also have a HwV380E12_WF9_PTZ_20180628 and it works with your [httpd_V380E2_C2.zip](https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_V380E2_C2_9090909_v20210620.zip)
The only difference I notice is that the prerun file is newer version and I see this message in the console: “ V380E login: 
====
prerun ver: V3.2.0_9aae6f3[build:Nov 11 2020_10:37:20]
====

*** to umount -l the sdcard ***
IFCReadStringOnce Warning:no such file(/mnt/sdcard/update_record)”

maybe it’s looking for a different file? On V380E2_C2 I don’t see this message. 
I also tried editing the header with the universal patch v380support tech sent me(which works), but no luck. If the patch works there must be a way. If you can think of anything, I’ll greatly appreciate it.

**Reply by Arkady23** [reply]
In the universal patch, the version number is 3030300. I.e. it will no longer work. If you make a new patch, then you need to adjust the version, for example, 3030301. If you want, you can take your patch and make a web server with the number 3030301. That is, leave everything unchanged, just add our addition there. But for some reason it seems to me that they made a mistake and gave you not your patch, not for your camera.
There is still such a moment. For example, the door eye camera has little memory. And if you run a lot of things, the web server will not be able to start. It is necessary to disable everything that can be disabled and leave only the web server. Just to check this one out.

**Reply by Geotv123** [reply]
I’m willing to try, where do I write 3030301? Which file do I edit?

**Reply by Arkady23** [reply]
[httpd_V380E3_C5_3030301_v20210620.zip](https://github.com/Arkady23/V380-Web/files/8658750/httpd_V380E3_C5_3030301_v20210620.zip)

**Reply by Geotv123** [reply]
> httpd_V380E3_C5_3030301_v20210620.zip

Thank you Arkady, now I see where the version number goes. Unfortunately the last file didn’t work

To clarify, the universal patch I have is for HwV380E22_WF6_CAT_PTZ_20160207
with soft_id V380E3_C5 version 3030209

Maybe I can try to download a patch for HwV380E21_WF3_PTZ_WIFI_20201212
soft_id V380E3_C5 version 3030300 from ...aliyuncs.com to compare

if I can get telnet to work it will help

**Comment by Arkady23** [comment]
Added a description for installing a web server manually for new cameras.
